setup/PowerShellSetup.Test/MsiLayoutTests.cs

```diff
@@ -0,0 +1,270 @@
+﻿namespace PowerShellSetup.Tests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Linq;
+    using System.Management.Automation;
+    using System.Threading.Tasks;
+    using Xunit;
+    using Xunit.Abstractions;
+
+    /// <summary>
+    /// Set of tests to run against MSI
+    /// These set of tests are especially for scenario like:
+    ///     1) Layout errors, things getting copied to the wrong location
+    ///     2) Checking if files in the MSI including the MSI are signed
+    ///     3) Checking if files included in MSI uses right signing alogirthm
+    /// 
+    /// </summary>
+    public class MsiLayoutTests: MsiTestBase
+    {
+        const string MSI_NAME = "AzurePowerShell.msi";
+        const string MSI_EXTRACT_DIR_NAME = "msiContents";
+        string _procOutput;
+        string _procErr;
+
+        public MsiLayoutTests(ITestOutputHelper testOutput) : base(testOutput)
+        {
```

Comment:
@markcowl in tasks, we filter files from our output, but we still add change files before we package it into msi. So the best verification would be to see what is signed and not signed inside the msi.
Every test extracts it once prior to running test.